### ISSUE TITLE
Tooltips placing from top to left (or other available space)

### ISSUE THREAD

**Issue by insanepl** [pivotal-cf/pivotal-ui#397]
I just noticed that your tooltips only allow to be placed in opposite directions.
top - bottom
left - right

Is there any way to make them fully 'bounce' from edges?
There's and example: http://react-component.github.io/tooltip/examples/placement.html

If tooltip inner text is too long to be placed on top because of left/right edge it should be placed on left or right then. Are you planning to develop such a feature like rc-tooltip?

Just in case attaching example, should be aligned to left in this case
![tooltip](https://cloud.githubusercontent.com/assets/1675802/15604919/2343aff8-2403-11e6-914e-5284c46516c4.jpg)

**Comment by charleshansen** [comment]
In short, there are no specific plans for the tooltip component. It already is doing more than anyone at Pivotal seems to need right now.

Our tooltip component is internally using tetherjs for placement. I believe tether supports all of the basic positioning you want from rc-tooltip. We haven't exposed that functionality yet because no one has needed it. It shouldn't be too hard so I would take a pr. Right now it takes a `placement` prop for the primary positioning, maybe something like `alignment` for secondary positioning could fill out the feature set.

We are also using constraints from tetherjs, so the tooltip shouldn't ever go off the page if the content is too long. It will re-position the tooltip when it goes off the page (unless you set `pin` false). I haven't tested all the cases yet, is the attached example using pivotal ui?

**Comment by insanepl** [comment]
Yes the example is from Pivotal UI.
That`s how it looks in Code

`return <OverlayTrigger
            placement="top"
            trigger="hover"
            delay="100"
            overlay={
            <Motion defaultStyle={{
                "opacity": 0...`

As you see, I`m using placement top and ofc it bounces from top to bottom when needed, but doesn`t bounce off right/left edges when there's no space.

I checked Tether and the effect I want to achieve is already here at "Constraints" Section => http://tether.io/

But seems it`s not working on Pivotal UI so would be great if they could implement same behavious since tether lib already got it.

**Comment by charleshansen** [comment]
I was able to reproduce the constraints issue. The constraints used to be `{to: 'window', attachment: 'together'}` but should actually be `{to: 'window', attachment: 'together', pin: true}`. It caused a weird positioning bug I also had to fix, but it should work. I will publish this soon.

I looked more carefully at the extra positioning in rc-tooltip and I don't think I'm going to implement them myself. There wasn't an obviously easy way doing it in tether and it will mostly involve a bunch of extra css. I think it should be possible to add the functionality yourself by adding custom classNames and css for them.
